Glossary and terms (#1264)
fishtown-analytics/docs.getdbt.com: website/docs/terms/view.md

---
id: view
title: View
displayText: view  
hoverSnippet: A view (as opposed to a table) is a defined passthrough SQL query that can be run against a database (or data warehouse).
---
:::important This page needs some love
This term lacks depth and requires more content. Would you like to contribute your knowledge to this page? [Create a discussion in our GitHub repository](https://github.com/dbt-labs/docs.getdbt.com/discussions) to begin the process of becoming a glossary contributor!
:::

A view (as opposed to a <Term id="table" />) is a defined passthrough SQL query that can be run against a database (or data warehouse). A view doesn’t store data, like a table does, but it defines the logic that you need to fetch the underlying data.

> For example, you might define a SQL view to count new users in a day as:

Reviewer: ```suggestion
For example, you might define a SQL view to count new users in a day:
```

```sql
  select
    created_date,
    count(distinct(user_id)) as new_users
  from users
  group by created_date
```

> That’s kind of tedious to write over and over again, so instead you could define it as a view called `new_users`, and instead query `select * from new_users`.

Reviewer: Just making sure people using screen readers know what you're referring to. 
```suggestion
But this SQL might get tedious to write over and over again, so instead you could define it as a view called `new_users`, and instead query `select * from new_users`.
```

When that `new_users` query runs, the underlying view compiles and runs against the database.  

## Tips on using views

A healthy relationship with views is built on expectations. 

You shouldn’t expect a view in itself to be your final destination in terms of data modeling (they’re slow + often more costly to query than tables, not great for connecting to a downstream process like reporting), but you should trust them to get you from point A to point B. 

> They’re like bus stops that you pass every day, never get out at, but appreciate as landmarks.

Reviewer: I love analogies but I'm not sure this one works for this example. Maybe bus stops you occasionally stop at when you forgot your coffee? Or the express bus vs the neighborhood line (not sure if this is a universal and global concept.)

Reviewer 2: Makes sense, removing the analogy
